sklearn/decomposition/pca.py

```diff
@@ -467,219 +502,11 @@ def score(self, X, y=None):
-
-        self.explained_variance_ = exp_var = (S ** 2) / n_samples
-        full_var = np.var(X, axis=0).sum()
-        self.explained_variance_ratio_ = exp_var / full_var
-
-        if self.whiten:
-            self.components_ = V / S[:, np.newaxis] * sqrt(n_samples)
-        else:
-            self.components_ = V
-
-        return X
-
-    def transform(self, X, y=None):
-        """Apply dimensionality reduction on X.
-
-        X is projected on the first principal components previous extracted
-        from a training set.
-
-        Parameters
-        ----------
-        X : array-like, shape (n_samples, n_features)
-            New data, where n_samples in the number of samples
-            and n_features is the number of features.
-
-        Returns
-        -------
-        X_new : array-like, shape (n_samples, n_components)
-
-        """
-        check_is_fitted(self, 'mean_')
-
-        X = check_array(X)
-        if self.mean_ is not None:
-            X = X - self.mean_
-
-        X = fast_dot(X, self.components_.T)
-        return X
-
-    def fit_transform(self, X, y=None):
-        """Fit the model with X and apply the dimensionality reduction on X.
-
-        Parameters
-        ----------
-        X : array-like, shape (n_samples, n_features)
-            New data, where n_samples in the number of samples
-            and n_features is the number of features.
-
-        Returns
-        -------
-        X_new : array-like, shape (n_samples, n_components)
-
-        """
-        X = check_array(X)
-        X = self._fit(X)
-        return fast_dot(X, self.components_.T)
-
-    def inverse_transform(self, X, y=None):
-        """Transform data back to its original space.
-
-        Returns an array X_original whose transform would be X.
-
-        Parameters
-        ----------
-        X : array-like, shape (n_samples, n_components)
-            New data, where n_samples in the number of samples
-            and n_components is the number of components.
-
-        Returns
-        -------
-        X_original array-like, shape (n_samples, n_features)
-
-        Notes
-        -----
-        If whitening is enabled, inverse_transform does not compute the
-        exact inverse operation of transform.
-        """
-        check_is_fitted(self, 'mean_')
-
-        X_original = fast_dot(X, self.components_)
-        if self.mean_ is not None:
-            X_original = X_original + self.mean_
-        return X_original
+@deprecated("it will be removed in 0.19. Use PCA(svd_solver='randomized') "
+            "instead. The new implementation DOES NOT store"
+            "whiten components_. Apply transform to get them.")
+def RandomizedPCA(n_components=None, copy=True, iterated_power=3,
+                  whiten=False, random_state=None):
+        return PCA(n_components=n_components, copy=copy, whiten=whiten,
+                   svd_solver='randomized', iterated_power=3,
+                   random_state=random_state)
```

Comment:
This change would not fix #3930, but I guess we do not care as we are deprecating anyway.